### DISCUSSION TITLE
[Application Development] L2-to-L1 messages stuck in IN_CHALLENGE_PERIOD

### DISCUSSION THREAD

**Discussion by paul-ilvez** [ethereum-optimism/developers#621] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi, new here, sorry if it's already been announced somewhere or raised in issue - couldn't find anything related.
I've tried to deploy my own crosschain L2-to-L1 application which exchanges data in that direction. 

I've followed this tutorial https://docs.optimism.io/builders/chain-operators/tutorials/sdk and used the @eth-optimism/sdk 3.1.2 (yes, I know it's dated) to prove my L2 Tx.

My crosschain message didn't get past the status 4 == IN_CHALLENGE_PERIOD.

Then I found out that no L2-to-L1 transaction got past the 'Waiting' status for quite a while. 
https://optimistic.etherscan.io/txsExit?ps=100&p=34
Is L2 to L1 programmatic messaging currently down for a while?

**Comment by imtipi** [comment]
> @eth-optimism/sdk 3.1.2 (yes, I know it's dated)

too dated,latest is 3.3.2(i wonder does 3.1.2 really updated for fault proof?)
also better try viem as optimism SDK discontinued
https://viem.sh/op-stack

**Reply by paul-ilvez** [reply]
Thanks for reply! But the core question is, do L2 to L1 programmatic messages actually work, with any of library? Or is it suspended/terminated for uses except token withdrawals?

**Reply by imtipi** [reply]
> Thanks for reply! But the core question is, do L2 to L1 programmatic messages actually work, with any of library? Or is it suspended/terminated for uses except token withdrawals?

of course it works, typically even token withdrawal is type of sending data , I suggest you use viem to do withdrawal
https://docs.optimism.io/builders/app-developers/bridging/messaging
https://docs.optimism.io/builders/notices/sdk-deprecation
